classes/gateways/class.pmprogateway_cybersource.php

```diff
@@ -513,8 +542,23 @@ function charge(&$order)
 			$item0->id = $order->membership_id;
 			$request->item = array($item0);
 
-			$soapClient = new CyberSourceSoapClient($wsdl_url, array("merchantID"=>pmpro_getOption("cybersource_merchantid"), "transactionKey"=>pmpro_getOption("cybersource_securitykey")));
-			$reply = $soapClient->runTransaction($request);
+			try
+			{
+				$soapClient = new CyberSourceSoapClient($wsdl_url, array("merchantID"=>$request->merchantID, "transactionKey"=>pmpro_getOption("cybersource_securitykey")));
+				$reply = $soapClient->runTransaction($request);
+			}
+			catch(Throwable $t)
+			{
+				$order->error = "Error communicating with Cybersource: " . $t->getMessage();
```

Comment:
When localizing these, we need to account for right to left languages. So something like

sprintf( __( 'Error communicating with Cybersource: %', 'paid-memberships-pro' ), $t->getMessage() );